### ISSUE TITLE
Implement month/day drop downs on home page and search results (catalog) page

### ISSUE THREAD

**Issue by kaylaondracek** [mlibrary/fishrappr#110]
## Description: 
As a user, I want to see more granular date range search on home page and on search (catalog) page so that I can approach research with a full date rather than by year.

## Acceptance Criteria:

- [x] 1. For search start and end, include Month & Day drop downs. Year is an open text field
- [x] 1.a. Validation: year should be ####
- [ ] 1.b. If user selects date that doesn't exist via drop downs (eg 2/31/2010), what is expected behavior? Error?
- [x] 2. Date search should be hidden initially

**Comment by njaffer** [comment]
Waiting on Trevor to do some wireframes for hidden date boxes

**Comment by gordonleacock** [comment]
**Notes from Gordon:**

1. No validation yet
2. Re AC 1.b.: I'll be able to fix this some the user can't select such a date. If they select February, I'll change the date drop-down to the allowed date values.
3. Re AC 2. Not done yet. Shows date range.

**Comment by gordonleacock** [comment]
**Added:**

1. Year validation; could still be improved?
2. Not handling months of less than 31 days yet.
3. BUG: In the box showing the results parameters, the range display can not be dismissed by clicking the "X"; instead you have to click start over.

**Comment by njaffer** [comment]
TODO later as a separate ticket:  If user selects date that doesn't exist via drop downs (eg 2/31/2010), what is expected behavior? Error?
